File: woodroid/src/main/java/me/gilo/woodroid/data/api/SettingsAPI.java
package me.gilo.woodroid.data.api;


import retrofit2.Call;
import me.gilo.woodroid.models.SettingGroup;
import me.gilo.woodroid.models.SettingOption;
import retrofit2.http.*;

import java.util.List;

public interface SettingsAPI {

    @GET("settings")
    Call<List<SettingGroup>> settings();

    @GET("settings/{group_id}/{id}")
    Call<SettingOption> option(@Path("group_id") String group_id, @Path("id") String option_id);

    @GET("settings/{id}")
    Call<List<SettingOption>> options(@Path("id") String group_id);

    @Headers("Content-Type: application/json")
    @PUT("settings/{group_id}/{id}")
    Call<SettingOption> update(
            @Path("group_id") String group_id,
            @Path("id") String option_id,
            @Body SettingOption body
    );



}

File: woodroid/src/main/java/me/gilo/woodroid/data/api/SettingsAPI.kt
package me.gilo.woodroid.data.api


import retrofit2.Call
import me.gilo.woodroid.models.SettingGroup
import me.gilo.woodroid.models.SettingOption
import retrofit2.http.*

interface SettingsAPI {

    @GET("settings")
    fun settings(): Call<List<SettingGroup>>

    @GET("settings/{group_id}/{id}")
    fun option(@Path("group_id") group_id: String, @Path("id") option_id: String): Call<SettingOption>

    @GET("settings/{id}")
    fun options(@Path("id") group_id: String): Call<List<SettingOption>>

    @Headers("Content-Type: application/json")
    @PUT("settings/{group_id}/{id}")
    fun update(
        @Path("group_id") group_id: String,
        @Path("id") option_id: String,
        @Body body: SettingOption
    ): Call<SettingOption>


}